spikeinterface/extractors/mcsh5extractors.py

```diff
@@ -0,0 +1,143 @@
+from spikeinterface import BaseRecording, BaseRecordingSegment
+import numpy as np
+from pathlib import Path
+
+try:
+    import h5py
+
+    HAVE_MCSH5 = True
+except ImportError:
+    HAVE_MCSH5 = False
+
+
+class MCSH5RecordingExtractor(BaseRecording):
+    """
+    Load a MCS H5 file as a recording extractor
+
+    Parameters
+    ----------
+    file_path : str or Path
+        The path to the MCS h5 file
+    stream_id : int, optional
+        The stream ID to load, by default 0
+        
+    Returns
+    -------
+    recording: MCSH5RecordingExtractor
+        The recording extractor for the MCS h5 file
+    """
+    extractor_name = 'MCSH5Recording'
+    installed = HAVE_MCSH5  # check at class level if installed or not
+    is_writable = False
+    mode = 'file'
+    installation_mesg = "To use the MCSH5RecordingExtractor install h5py: \n\n pip install h5py\n\n"  # error message when not installed
+
+    def __init__(self, file_path, stream_id=0):
+        assert self.installed, self.installation_mesg
+        self._file_path = file_path
+
+        mcs_info = openMCSH5File(self._file_path, stream_id)
+        self._rf = mcs_info["filehandle"]
+
+        BaseRecording.__init__(self, sampling_frequency=mcs_info["sampling_frequency"], channel_ids=mcs_info["channel_ids"],
+                               dtype=mcs_info["dtype"])
+        
+        recording_segment = MCSH5RecordingSegment(self._rf, stream_id, mcs_info["num_frames"], 
+                                                  sampling_frequency=mcs_info["sampling_frequency"])
+        self.add_recording_segment(recording_segment)
+        
+        # set gain
+        self.set_channel_gains(mcs_info["gain"])
+        
+        # set other properties
+        self.set_property("electrode_labels", mcs_info["electrode_labels"])
+        
+        self._kwargs = {'file_path': str(Path(file_path).absolute()), 'stream_id': stream_id}
+
+    def __del__(self):
+        self._rf.close()
+
+
+class MCSH5RecordingSegment(BaseRecordingSegment):
+    
+    def __init__(self, rf, stream_id, num_frames, sampling_frequency):
+        BaseRecordingSegment.__init__(self, sampling_frequency=sampling_frequency)
+        self._rf = rf
+        self._stream_id = stream_id
+        self._num_samples = int(num_frames)
+
+    def get_num_samples(self):
+        return self._num_samples
+    
+    def get_traces(self, 
+                   start_frame=None, 
+                   end_frame=None, 
+                   channel_indices=None):
+        stream = self._rf.require_group('/Data/Recording_0/AnalogStream/Stream_' + str(self._stream_id))
```

Comment:
This should be done in init no ?
So avoid to fetch it at each chunk.

Reply:
yep! Good catch! Fixing it now :)